### PR TITLE
Handle empty string when converting external reference values 

### PR DESCRIPTION
**Pull request by aaclan-ebi**
Fixes HumanCellAtlas/ingest-central#533

### CODE REVIEW

File: ingest/importer/conversion/conversion_strategy.py [hunk 1]
```diff
@@ -125,8 +125,9 @@ def __init__(self, field, main_category):
         self.main_category = main_category
 
     def apply(self, metadata: MetadataEntity, cell_data):
-        link_ids = self.converter.convert(cell_data)
-        metadata.add_external_links(self.main_category, link_ids)
+        if cell_data is not None and cell_data != '':
```
Comment (rdgoite) [inline ingest/importer/conversion/conversion_strategy.py]: I think `if cell_data:` will work similarly here as both `None` and `''` have `False` truthiness.

File: ingest/importer/importer.py [hunk 2]
```diff
@@ -25,12 +25,12 @@ def __init__(self, ingest_api):
...
 
     def dry_run_import_file(self, file_path, project_uuid=None):
-        spreadsheet_json, template_mgr, errors = self._generate_spreadsheet_json(file_path, project_uuid)
+        spreadsheet_json, template_mgr, errors = self.generate_json(file_path, project_uuid)
         entity_map = self._process_links_from_spreadsheet(template_mgr, spreadsheet_json)
 
         return entity_map
 
-    def _generate_spreadsheet_json(self, file_path, project_uuid=None):
+    def generate_json(self, file_path, project_uuid=None):
```
Comment (rdgoite) [inline ingest/importer/importer.py]: Why has `generate_json` been promoted to public? Do we have tests for this unit?

File: tests/importer/test_importer.py [hunk 3]
```diff
@@ -312,7 +312,7 @@ def test_import_file_throwing_exception(self):
         exception = Exception('Error thrown for Unit Test')
         exception_json = ImporterError(str(exception)).getJSON()
         importer = XlsImporter(ingest_api=self.mock_ingest_api)
-        importer._generate_spreadsheet_json = MagicMock(side_effect=exception)
+        importer.generate_json = MagicMock(side_effect=exception)
```
Comment (rdgoite) [inline tests/importer/test_importer.py]: If we're testing the importer, we really shouldn't be mocking any part of it.

Comment (aaclan-ebi) [reply tests/importer/test_importer.py]: it's testing the throwing of error in import_file. =

File: ingest/importer/importer.py (hunk 2 above)
Comment (aaclan-ebi) [reply ingest/importer/importer.py]: Added.

File: ingest/importer/conversion/conversion_strategy.py (hunk 1 above)
Comment (aaclan-ebi) [reply ingest/importer/conversion/conversion_strategy.py]: Makes sense! Thanks!

**Pull request merged**